include/spdlog/details/circular_q.h

```diff
@@ -72,6 +72,27 @@ class circular_q
         return v_[head_];
     }
 
+    // Return number of elements actually stored
+    size_t size() const
+    {
+        if (tail_ > head_)
+        {
+           return (tail_ - head_) % max_items_;
+        }
+        else
+        {
+            return max_items_ - (head_ - tail_ ) % max_items_;
+        }
+    }
```

Comment:
no need to % max_items and will give wrong result if if tail_ == head_(i.e. if empty it will give max_items_)

Here is a fixed version:

size_t size() const
    {
        if (tail_ >= head_)
        {
           return (tail_ - head_);
        }
        else
        {
            return max_items_ - (head_ - tail_ );
        }
    }

Reply:
Thanks for patience. Fixed.